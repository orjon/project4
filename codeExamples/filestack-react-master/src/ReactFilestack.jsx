import React, { Component } from 'react';
import * as filestack from 'filestack-js';
import PropTypes from 'prop-types';

class ReactFilestack extends Component {
  static defaultProps = {
    file: null,
    link: false,
    buttonText: 'Pick file',
    buttonClass: '',
    onSuccess: result => console.log(result),
    onError: error => console.error(error),
    mode: 'pick',
    options: {},
    security: null,
    children: null,
    render: null,
    cname: null,
    sessionCache: false,
    preload: false,
  };

  static propTypes = {
    file: PropTypes.objectOf(PropTypes.any),
    apikey: PropTypes.string.isRequired,
    link: PropTypes.bool,
    mode: PropTypes.string,
    buttonText: PropTypes.string,
    buttonClass: PropTypes.string,
    onSuccess: PropTypes.func,
    onError: PropTypes.func,
    options: PropTypes.objectOf(PropTypes.any),
    security: PropTypes.objectOf(PropTypes.any),
    children: PropTypes.node,
    render: PropTypes.func,
    cname: PropTypes.string,
    sessionCache: PropTypes.bool,
    preload: PropTypes.bool,
  };

  constructor(props) {
    super(props);
    const {
      apikey,
      security,
      cname,
      sessionCache,
      preload,
      options,
    } = this.props;
    const client = filestack.init(apikey, {
      security,
      cname,
      sessionCache,
    });
    this.state = {
      client,
      picker: preload ? client.picker({ ...options, onUploadDone: this.onFinished }) : null,
    };

    this.onFinished = this.onFinished.bind(this);
    this.onFail = this.onFail.bind(this);
  }

  onClickPick = (event) => {
    event.stopPropagation();
    event.preventDefault();

    const {
      client,
      picker,
    } = this.state;

    const {
      options,
      mode,
      file,
      security,
      preload,
    } = this.props;

    this.callPicker(mode, options, file, security, preload, client, picker)
      .then(this.onFinished)
      .catch(this.onFail);
  };

  onFinished = (result) => {
    const { onSuccess } = this.props;
    if (typeof onSuccess === 'function' && result) {
      onSuccess(result);
    }
  };

  onFail = (error) => {
    const { onError } = this.props;
    if (typeof onError === 'function') {
      onError(error);
    } else {
      console.error(error);
    }
  };

  callPicker = (mode, options, file, security, preload, client, picker) => {
    const { url, handle } = options;
    delete options.handle;
    delete options.url;

    if (mode === 'transform') {
      return new Promise((resolve, reject) => {
        try {
          resolve(client.transform(handle, options));
        } catch (err) {
          reject(err);
        }
      });
    } else if (mode === 'retrieve') {
      return client.retrieve(handle, options);
    } else if (mode === 'metadata') {
      return client.metadata(handle, options);
    } else if (mode === 'storeUrl') {
      return client.storeURL(url, options);
    } else if (mode === 'upload') {
      return client.upload(file, options);
    } else if (mode === 'remove') {
      return client.remove(handle, security);
    }

    return new Promise((resolve) => {
      if (preload) {
        picker.open();
        resolve();
      } else {
        client.picker({ ...options, onUploadDone: resolve }).open();
      }
    });
  };

  render () {
    const {
      buttonClass, buttonText, link, children, render: CustomRender,
    } = this.props;
    if (CustomRender) {
      return (
        <CustomRender onPick={this.onClickPick} />
      );
    }
    const Tag = link ? 'a' : 'button';
    return (
      <Tag
        name="filestack"
        onClick={this.onClickPick}
        className={buttonClass}
      >
        {children || buttonText}
      </Tag>
    );
  }
}

export default ReactFilestack;
