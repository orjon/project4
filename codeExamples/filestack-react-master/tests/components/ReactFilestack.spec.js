import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import * as filestack from 'filestack-js';
import ReactFilestack from '../../src/ReactFilestack';

configure({ adapter: new Adapter() });

const test = () => <div>test</div>; // eslint-disable-line

console = {
  log: jest.fn(),
  error: jest.fn(),
};

const pickerMock = jest.fn(() => ({
  open: jest.fn(),
}));

jest.spyOn(filestack, 'init').mockImplementation(() => ({
  picker: pickerMock,
  upload: jest.fn(() => Promise.resolve()),
  metadata: jest.fn(() => Promise.resolve()),
  remove: jest.fn(() => Promise.resolve()),
  storeURL: jest.fn(() => Promise.resolve()),
  retrieve: jest.fn(() => Promise.resolve()),
}));

const createWrapper = (props = {}) => {
  props.apikey = 'Acu94EFL1STGYvkM6a8usz';
  return shallow(
    <ReactFilestack {...props} />,
  );
};

let props;
let wrapper;

describe('<ReactFilestack />', () => {
  afterEach(() => {
    pickerMock.mockClear();
  });

  it('should render ReactFilestack component', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should render the component', () => {
    props = {
      onSuccess: (result) => {
        console.log('result', result);
      },
      onError: (error) => {
        console.error('error', error);
      },
      render: test,
    };
    wrapper = createWrapper(props);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should run pick as default', () => {
    wrapper = createWrapper();
    wrapper.find('button').simulate('click', { stopPropagation () {}, preventDefault() {} });
    expect(wrapper).toMatchSnapshot();
  });

  it('should not allow wrong option properties', () => {
    props = {
      options: { wrong: 'handle' },
    };
    wrapper = createWrapper(props);
    wrapper.find('button').simulate('click', { stopPropagation () {}, preventDefault() {} });
    expect(wrapper).toMatchSnapshot();
  });

  it('should run metadata and mock handle', () => {
    props = {
      mode: 'metadata',
      options: { handle: 'handle' },
    };
    wrapper = createWrapper(props);
    wrapper.find('button').simulate('click', { stopPropagation () {}, preventDefault() {} });
    expect(wrapper).toMatchSnapshot();
  });

  it('should run remove width mock handle and security', () => {
    props = {
      mode: 'remove',
      options: { handle: 'handle' },
      security: { policy: 'policy', signature: 'signature' },
    };
    wrapper = createWrapper(props);
    wrapper.find('button').simulate('click', { stopPropagation () {}, preventDefault() {} });
    expect(wrapper).toMatchSnapshot();
  });

  it('should run storeUrl with mock url', () => {
    props = {
      mode: 'storeUrl',
      options: { url: 'url' },
    };
    wrapper = createWrapper(props);
    wrapper.find('button').simulate('click', { stopPropagation () {}, preventDefault() {} });
    expect(wrapper).toMatchSnapshot();
  });

  it('should run retrieve with mock handle', () => {
    props = {
      mode: 'retrieve',
      options: { handle: 'handle' },
    };
    wrapper = createWrapper(props);
    wrapper.find('button').simulate('click', { stopPropagation () {}, preventDefault() {} });
    expect(wrapper).toMatchSnapshot();
  });

  it('should run transform with mock url', () => {
    props = {
      mode: 'transform',
      options: { url: 'url' },
    };
    wrapper = createWrapper(props);
    wrapper.find('button').simulate('click', { stopPropagation () {}, preventDefault() {} });
    expect(wrapper).toMatchSnapshot();
  });

  it('should throw an exception with transform and wrong options properties', () => {
    props = {
      mode: 'transform',
      options: { handle: 'abc123', wrong: 'wrong' },
    };
    wrapper = createWrapper(props);
    wrapper.find('button').simulate('click', { stopPropagation () {}, preventDefault() {} });
    expect(wrapper).toMatchSnapshot();
  });

  it('should run upload with mock data', () => {
    props = {
      mode: 'upload',
      options: { url: 'url' },
      file: new Blob([JSON.stringify({ hello: 'world' }, null, 2)], { type: 'application/json' }),
    };
    wrapper = createWrapper(props);
    wrapper.find('button').simulate('click', { stopPropagation () {}, preventDefault() {} });
    expect(wrapper).toMatchSnapshot();
  });

  it('should create picker instance before click on the button when preload true', () => {
    props = {
      preload: true,
    };
    wrapper = createWrapper(props);
    expect(pickerMock).toHaveBeenCalledTimes(1);
    wrapper.find('button').simulate('click', { stopPropagation () {}, preventDefault() {} });
    expect(pickerMock).toHaveBeenCalledTimes(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('should create picker instance after click on the button when preload false(default)', () => {
    wrapper = createWrapper();
    expect(pickerMock).toHaveBeenCalledTimes(0);
    wrapper.find('button').simulate('click', { stopPropagation () {}, preventDefault() {} });
    expect(pickerMock).toHaveBeenCalledTimes(1);
    expect(wrapper).toMatchSnapshot();
  });
});
