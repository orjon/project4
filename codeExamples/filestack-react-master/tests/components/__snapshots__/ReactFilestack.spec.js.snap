// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReactFilestack /> should create picker instance after click on the button when preload false(default) 1`] = `ShallowWrapper {}`;

exports[`<ReactFilestack /> should create picker instance before click on the button when preload true 1`] = `ShallowWrapper {}`;

exports[`<ReactFilestack /> should not allow wrong option properties 1`] = `ShallowWrapper {}`;

exports[`<ReactFilestack /> should render ReactFilestack component 1`] = `null`;

exports[`<ReactFilestack /> should render the component 1`] = `
<test
  onPick={[Function]}
/>
`;

exports[`<ReactFilestack /> should run metadata and mock handle 1`] = `ShallowWrapper {}`;

exports[`<ReactFilestack /> should run pick as default 1`] = `ShallowWrapper {}`;

exports[`<ReactFilestack /> should run remove width mock handle and security 1`] = `ShallowWrapper {}`;

exports[`<ReactFilestack /> should run retrieve with mock handle 1`] = `ShallowWrapper {}`;

exports[`<ReactFilestack /> should run storeUrl with mock url 1`] = `ShallowWrapper {}`;

exports[`<ReactFilestack /> should run transform with mock url 1`] = `ShallowWrapper {}`;

exports[`<ReactFilestack /> should run upload with mock data 1`] = `ShallowWrapper {}`;

exports[`<ReactFilestack /> should throw an exception with transform and wrong options properties 1`] = `ShallowWrapper {}`;
